### PR TITLE
Elpp 2495 - ConvertImagesToJPG extracted from ResizeImages

### PR DESCRIPTION
**Pull request by Jenniferstrej**
with content refactored into a provider.
ConvertImagesToJPG uses a provider and more StorageProvider functions. It's basically a refactored ResizeImages and excludes some formats (originally from formats.yaml)

### CODE REVIEW

File: activity/activity_ConvertImagesToJPG.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+            storage_context = StorageContext(self.settings)
+            files_in_bucket = storage_context.list_resources(orig_resource)
+
+            figures = filter(article_structure.article_figure, files_in_bucket)
+
+            formats = {"Original": {
+                            "sources": "tif",
+                            "format": "jpg",
+                            "resolution": 96,
```
Comment (giorgiosironi) [inline activity/activity_ConvertImagesToJPG.py]: this `96` looks more like a `dpi` parameter than a `resolution` like `800x600`

File: activity/activity_ConvertImagesToJPG.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                add_resource_path = storage_provider + published_bucket_path + "/" + article_id + "/"
+
+                publish_locations = [cdn_resource_path, add_resource_path]
+
+                image_conversion.generate_images(self.settings, formats, file_pointer, article_structure.ArticleInfo(file_name),
+                                                 publish_locations, self.logger)
+
+            self.emit_monitor_event(self.settings, article_id, version, run, self.pretty_name, "end",
+                                    "Finished converting images for " + article_id + ": " +
+                                    str(len(files_in_bucket)) + " images processed ")
```
Comment (giorgiosironi) [inline activity/activity_ConvertImagesToJPG.py]: only `len(figures)` were actual images to be processed right?

File: activity/activity_ConvertImagesToJPG.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            storage_context = StorageContext(self.settings)
+            files_in_bucket = storage_context.list_resources(orig_resource)
+
+            figures = filter(article_structure.article_figure, files_in_bucket)
+
+            formats = {"Original": {
+                            "sources": "tif",
+                            "format": "jpg",
+                            "resolution": 96,
+                            "download": "yes"
```
Comment (giorgiosironi) [inline activity/activity_ConvertImagesToJPG.py]: I don't think we need the `download` version, in the 2.0 case; the images created here will be read only by the IIIF server

File: workflow/workflow_IngestArticleZip.py [hunk 4]
```diff
@@ -90,6 +90,17 @@ def __init__(self, settings, logger, conn=None, token=None, decision=None,
                         "start_to_close_timeout": 60 * 15
                     },
                     {
+                        "activity_type": "ConvertImagesToJPG",
```
Comment (giorgiosironi) [inline workflow/workflow_IngestArticleZip.py]: :+1: right place

File: provider/image_conversion.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+                format_spec = formats[format_spec_name]
+                # if sources not present or includes file extension for this image
+                if 'sources' not in format_spec or info.extension in [
+                        x.strip() for x in format_spec['sources'].split(',')]:
+                    download = 'download' in format_spec and format_spec['download']
+                    fp.seek(0)  # rewind the tape
+                    filename, image = resizer.resize(format_spec, fp, info, logger)
+                    if filename is not None and image is not None:
+                        store_in_publish_locations(settings, filename, image, publish_locations, download)
+                        logger.info("Stored image %s as %s" % (filename, str(publish_locations)))
```
Comment (giorgiosironi) [inline provider/image_conversion.py]: looks like if `filename` or `image` are `None` we have encountered a error; throwing an exception?

File: activity/activity_ConvertImagesToJPG.py (hunk 3 above)
Comment (giorgiosironi) [inline activity/activity_ConvertImagesToJPG.py]: (the provider may support it, but we may not need to actually require it from this activity)

File: activity/activity_ConvertImagesToJPG.py (hunk 2 above)
Comment (Jenniferstrej) [inline activity/activity_ConvertImagesToJPG.py]: True. In ResizeImages I believe there is no such filter. I'll change.

File: activity/activity_ConvertImagesToJPG.py (hunk 3 above)
Comment (Jenniferstrej) [inline activity/activity_ConvertImagesToJPG.py]: Ok, I'll remove it from the formats definition only

File: activity/activity_ConvertImagesToJPG.py (hunk 1 above)
Comment (Jenniferstrej) [inline activity/activity_ConvertImagesToJPG.py]: I wonder where this "resolution" is used...

Comment (Jenniferstrej) [inline activity/activity_ConvertImagesToJPG.py]: I'm removing it altogether from the formats dictionary. It doesn't seem to influence anything. Less confusing.

**Pull request merged**